services/core/MasterDriverAgent/master_driver/interfaces/modbus_tk/config_cmd.py

```diff
@@ -0,0 +1,947 @@
+                    "interval": interval,
+                    "timezone": "UTC"
+                }
+            else:
+                driver_config = {
+                    "driver_config": {"name": name,
+                                      "device_type": device_type_name,
+                                      "device_address": device_address,
+                                      "port": port,
+                                      "addressing": addressing,
+                                      "endian": endian,
+                                      "write_multiple_registers": write_multiple_registers,
+                                      "description": description},
+                    "driver_type": "modbus_tk",
+                    "registry_config": "config://" + csv_file,
+                    "interval": interval,
+                    "timezone": "UTC"
+                }
+
+            # RTU transport
+            if not port:
+                print('Enter slave id (default to 1): ', end='')
+                try:
+                    slave_id = int(raw_input())
+                except ValueError:
+                    slave_id = 1
+
+                print('Enter baudrate (default to 9600): ', end='')
+                try:
+                    baudrate = int(raw_input())
+                except ValueError:
+                    baudrate = 9600
+
+                print('Enter bytesize (default to 8): ', end='')
+                try:
+                    bytesize = int(raw_input())
+                except ValueError:
+                    bytesize = 8
+
+                print('Enter bytesize (default to none): ', end='')
+                parity = raw_input()
+                if parity not in ('none', 'even', 'odd', 'mark', 'space'):
+                    parity = 'none'
+
+                print('Enter stopbits (default to 1): ', end='')
+                try:
+                    stopbits = int(raw_input())
+                except ValueError:
+                    stopbits = 1
+
+                print('Enter xonxoff (default to 0): ', end='')
+                try:
+                    xonxoff = int(raw_input())
+                except ValueError:
+                    xonxoff = 0
+
+                driver_config['driver_config'].update({
+                    "slave_id": slave_id,
+                    "baudrate": baudrate,
+                    "bytesize": bytesize,
+                    "parity": parity,
+                    "stopbits": stopbits,
+                    "xonxoff": xonxoff
+                })
+
+            with open("{0}/{1}.config".format(self._directories['config_dir'], name), 'w') as config_file:
+                json.dump(driver_config, config_file, indent=2)
+
+        else:
+            print("Device type '{0}' does not exist".format(device_type_name))
+
+    ##########################
+    #    VOLTTRON Commands   #
+    ##########################
+
+    def do_load_volttron(self, line):
```

Comment:
We already have utilities for automatically adding and extracting configuration files from the master driver. This seems redundant.

Reply:
config_cmd is not only about adding and extracting configuration files from the master driver, but also manually adding/editing modbus-tk driver configuration by going through every parameters such as: name, device_type, device_address, port, slave_id, etc. The load_volttron is a part of config_cmd in which the program will look up all .config and .csv files in given directories, if the files does not exist, user can have a choice to either choose another file, change directory, or add the file to the selected directory.

If this is redundant, can you please point me to where the code is? Thanks.